### ISSUE TITLE
Faker adds path objects to sys.path_importer_cache, breaking other packages

### ISSUE THREAD

**Issue by mgorny** [joke2k/faker#1421]
* Faker version: 6.6.3
* OS: Gentoo Linux

After importing `faker`, entries with `PosixPath` objects are added as keys to `sys.path_importer_cache`. However, the keys are supposed to be regular `str`s there, and the wrong type breaks software relying on `str` methods being available, e.g. astroid:

```
___________________________________________ ClassNodeTest.test_slots_added_dynamically_still_inferred ____________________________________________

self = <tests.unittest_scoped_nodes.ClassNodeTest testMethod=test_slots_added_dynamically_still_inferred>

    def tearDown(self):
        del sys.path[0]
        datadir = find("")
        for key in list(sys.path_importer_cache):
>           if key.startswith(datadir):
E           AttributeError: 'PosixPath' object has no attribute 'startswith'

tests/resources.py:41: AttributeError
```

Note that since Faker installs a pytest plugin, it is autoloaded by default in all programs' test suites.

### Steps to reproduce

```
import sys
import faker
print(sys.path_importer_cache)
```

### Expected behavior

The printed dict should only contain `str` keys.

### Actual behavior

```
[...] PosixPath('/usr/lib/python3.9/site-packages/faker/providers/address'): FileFinder(PosixPath('/usr/lib/python3.9/site-packages/faker/providers/address')), PosixPath('/usr/lib/python3.9/site-packages/faker/providers/automotive'): FileFinder(PosixPath('/usr/lib/python3.9/site-packages/faker/providers/automotive')), [...]
```

**Comment by fcurella** [comment]
Thank you for the repot @mgorny ! Would you mind submitting a Pull Request?

**Comment by mgorny** [comment]
I actually hoped you know the code better and know where these paths are added. If you can tell me that, I can try. However, I'm really short on time these days.

**Comment by fcurella** [comment]
@mgorny

I think the root of the problem is [this line](https://github.com/joke2k/faker/blob/master/faker/utils/loading.py#L25), where we're returning a `Path` objects instead of a string.

I can write the Pull Request, but I'd like for you to try my changes to see if they fix your issue.

**Comment by levrik** [comment]
@fcurella We've started seeing this issue today the first time without any code changes or upgrades on Faker.
The only change was the upgrade from Python 3.8.9 to 3.8.10 after which this issue started to happen.

**Comment by CheesyFeet** [comment]
Can confirm pinning Python to 3.8.9 was a workaround for us too

**Comment by CheesyFeet** [comment]
After looking into it, we were running Faker 6.6.3 which didn't include the patch referenced by this issue. Pinning Faker to 8.1.2 fixed this for us on Python 3.8.10 (previously we didn't have Faker pinned as we're using it as a peer dependency of [Factory Boy](https://github.com/FactoryBoy/factory_boy))
